packages/main/src/Calendar.ts

```diff
@@ -52,6 +52,12 @@ interface ICalendarPicker {
 	_lastYear?: number,
 }
 
+/**
+ * Interface for components that may be slotted inside a `ui5-calendar`.
+ *
+ * **Note:** Use with `ui5-date` or `ui5-date-range` as calendar date selection types. Implementing the interface does not guarantee that any other class can work as a calendar date selection type.
```

Comment:
IMO just remove the "Implementing..." sentence. It sounds confusing. It's very unlikely somebody would try to implement their own dates and expect them to work out of the box purely due to having implemented this interface.